Resources/Private/BackendModule/src/Components/FocusKeywordListItem.tsx

```diff
@@ -0,0 +1,181 @@
+import PropTypes from "prop-types";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faCheck, faSpinner, faExternalLinkAlt} from "@fortawesome/free-solid-svg-icons";
+import React from "react";
+
+@connect(null, (dispatch, ownProps) => ({
+    update(propertyValue: string) {
+        dispatch(updateItemProperty({ identifier: ownProps.item.identifier, propertyName: 'focusKeyword', propertyValue }))
+    },
+    persist() {
+        dispatch(persistOneItem(ownProps.item.identifier))
+    },
+    generate() {
+        dispatch(generateItem(ownProps.item))
+    }
+}))
+export default class FocusKeywordListItem extends PureComponent {
+    static propTypes = {
+        item: PropTypes.object.isRequired,
+        update: PropTypes.func,
+        persist: PropTypes.func,
+        generate: PropTypes.func
+    }
+    private readonly iframeRef;
+
+    constructor(props) {
+        super(props);
+        this.iframeRef = React.createRef();
+    }
+
+    handleChange(event) {
+        const {update} = this.props;
+        update(event.target.value)
+    }
+
+    discard(): void
+    {
+        const {update} = this.props;
+        update('')
+    }
+
+    canChangeValue(): boolean
+    {
+        const {item} = this.props;
+        return !(item.persisted || item.generating || item.persisting);
+    }
+
+    canDiscard(): boolean
+    {
+        const {item} = this.props;
+        return !(item.persisted || item.generating || item.persisting || item.focusKeyword === '');
+    }
+
+    canPersist(): boolean
+    {
+        const {item} = this.props;
+        return !(item.persisted || item.generating || item.persisting || item.focusKeyword === '');
+    }
+
+    canGenerate(): boolean
+    {
+        const {item} = this.props;
+        return !(item.persisted || item.generating || item.persisting);
+    }
+
+    saveButtonLabel() {
+        const {item} = this.props;
+        if (item.persisting) {
+            return (
+                <span>
+                    <FontAwesomeIcon icon={faSpinner} spin={true}/>
+                    &nbsp;
+                    {this.translationService.translate('NEOSidekick.AiAssistant:Module:persisting', 'Saving...')}
+                </span>
+            )
+        } else if (item.persisted) {
+            return (
+                <span>
+                    <FontAwesomeIcon icon={faCheck} />
+                    &nbsp;
+                    {this.translationService.translate('NEOSidekick.AiAssistant:Module:persisted', 'Saved')}
+                </span>
+            )
+        } else {
+            return (
+                <span>
+                    {this.translationService.translate('NEOSidekick.AiAssistant:Module:persist', 'Save')}
+                </span>
+            )
+        }
+    }
+
+    render() {
+        const { item, persist, generate } = this.props;
+        const textfieldId = item.identifier
+        if (item.pageContent) {
+            const iframe = this.iframeRef.current
```

Comment:
Bei jedem render schreibst du den iframe inhalt neu. Ich würd vermuten, wenn du im iFrame scrollst und dann was generieren lasst springt das iFrame. Ich würd das in eine didUpdate schreiben, und zwar nur wenn sich der pageContent geändert hat.

Reply:
Hab es gerade probiert: wenn ich generiere oder speichere bleibt die scroll position komplett gleich. Ich nehme dann mal an, dass es keine Auswirkungen hat.